src/pages/api/frontend/exercise-submission.ts

```diff
@@ -0,0 +1,48 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+import { prisma } from '@/helper/prisma'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== 'POST') {
+    res.status(405).send({ message: 'Only POST requests allowed' })
+    return
+  }
+
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const { path, entityId, revisionId, type, result, sessionId } = req.body
+
+    if (
```

Comment:
you could also use an `io-ts` decoder here since we already have that dep. in the frontend.
([example](https://github.com/serlo/frontend/blob/staging/src/edtr-io/plugins/page-team/editor.tsx#L9))

Reply:
good hint